selfservice/flow/verification/handler.go

```diff
@@ -404,7 +413,7 @@ func (h *Handler) submitFlow(w http.ResponseWriter, r *http.Request, ps httprout
 		return
 	}
 
-	if f.Type == flow.TypeBrowser && !x.IsJSONRequest(r) {
+	if x.IsBrowserRequest(r) {
```

Comment:
We have multiple instances in the E2E tests, where a verification flow is created via the API and then submitted via the browser. Here, we make sure that every browser request is redirected to the UI.